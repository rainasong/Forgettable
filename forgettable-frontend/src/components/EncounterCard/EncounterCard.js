/* eslint-disable max-len */
import * as React from 'react';
import Card from '@mui/material/Card';
import CustomButton from '../CustomButton/CustomButton';
import {AvatarGroup, Avatar} from '@mui/material';
import classes from './EncounterCard.module.css';
import stringAvatar from '../../functions/helper';

const DELETE = 'Delete';
const DATE_WE_MET = 'Date we met: ';
const WHERE_WE_MET = 'Where we met: ';

export default function EncounterCard(props) {
  const {isInitialEncounter, title, persons, details, date, location, onClick, onDelete} = props;
  const isMultiplePerson = persons.length > 1;

  return (
    <Card className={isInitialEncounter ? classes.Card_special : classes.Card}>
      <div className={classes.Card_content}>
        <div onClick={onClick}>
          {isInitialEncounter && <section className={classes.Initial_label_container}>
            <div className={classes.Initial_label}>First Encounter!</div>
          </section>}
          <section className={classes.Header}>
            <div className={classes.Title}>
              {title}
            </div>
            <div className={classes.Profile_container}>
              <AvatarGroup max={4} className={isMultiplePerson ? classes.Avatar_multiple : classes.Avatar_inline}>
                {persons.map((person, index) => {
                  return (
                    <div key={index}>
                      <Avatar key={index} className={classes.Avatar} {...stringAvatar(`${person.first_name} ${person.last_name}`)} />
                      {!isMultiplePerson && <div className={classes.Avatar_name}>
                        {person.first_name}
                      </div>}
                    </div>
                  );
                },
                )}
              </AvatarGroup>
            </div>
          </section>
          <section className={classes.Body}>
            {details}
          </section>
        </div>
        <section className={classes.Footer}>
          <div className={classes.Date}>
            {DATE_WE_MET}<div>{date}</div>
          </div>
          <div className={classes.Location}>
            {WHERE_WE_MET}<div>{location}</div>
          </div>
          <CustomButton className={classes.Button} btnText={DELETE} onClick={onDelete}/>
        </section>
      </div>
    </Card>
  );
}
