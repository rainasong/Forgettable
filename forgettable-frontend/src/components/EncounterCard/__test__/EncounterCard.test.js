import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import EncounterCard from '../EncounterCard';

const isInitialEncounter = true;
const persons = [
  {first_name: 'Kent', last_name: 'Dodds'},
  {first_name: 'Jed', last_name: 'Watson'},
  {first_name: 'Kent', last_name: 'Dodds'},
  {first_name: 'Jed', last_name: 'Watson'},
  {first_name: 'Kent', last_name: 'Dodds'},
  {first_name: 'Jed', last_name: 'Watson'},
];
const handleCardOnClick = () => {
  console.log('card click!');
};

const handleDelete = () => {
  console.log('delete');
};

it('renders EncounterCard UI to test correct structure', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<EncounterCard
    isInitialEncounter={isInitialEncounter}
    title={'Fermentum pellentesque Fermentum pellentesque'}
    persons={persons}
    details={'Diam dictum vestibulum mi nulla vestibulum, id nibh. Nunc consequat amet commodo turpis tellus. Scelerisque a pellentesque vel accumsan sed mauris, ac turpis pharetra. Sem tristique nulla cursussssss praesent tincidunt integer'}
    date={'13/01/1872'}
    location={'Atlantis'}
    onClick={handleCardOnClick}
    onDelete={handleDelete} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
