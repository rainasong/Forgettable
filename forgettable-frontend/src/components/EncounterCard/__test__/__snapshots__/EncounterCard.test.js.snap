// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EncounterCard UI to test correct hierarchy 1`] = `
<div
  className="Card"
  onClick={[Function]}
>
  <div
    className="Card_content"
  >
    <section
      className="Header"
    >
      <div
        className="Title"
      >
        Example Title
      </div>
      <div
        className="Profile_container"
      >
        <CustomAvatar
          persons={
            Array [
              Object {
                "first_name": "Kent",
                "last_name": "Dodds",
              },
              Object {
                "first_name": "Jed",
                "last_name": "Watson",
              },
              Object {
                "first_name": "Marley",
                "last_name": "George",
              },
              Object {
                "first_name": "Jack",
                "last_name": "Winston",
              },
              Object {
                "first_name": "John",
                "last_name": "Doe",
              },
            ]
          }
        />
      </div>
    </section>
    <section
      className="Body"
    >
      Example Description
    </section>
    <section
      className="Footer"
    >
      <div
        className="Date"
      >
        Date we met: 
        <div>
          <UnknownDetail />
        </div>
      </div>
      <div
        className="Location"
      >
        Where we met: 
        <span>
          Example Location
        </span>
      </div>
      <CustomButton
        btnText="Delete"
        onClick={[Function]}
      />
    </section>
  </div>
</div>
`;
